Assets/Scripts/Models/HeadlineGenerator.cs

```diff
@@ -0,0 +1,69 @@
+#region License
+// ====================================================
+// Project Porcupine Copyright(C) 2016 Team Porcupine
+// This program comes with ABSOLUTELY NO WARRANTY; This is free software, 
+// and you are welcome to redistribute it under certain conditions; See 
+// file LICENSE, which is part of this source code package, for details.
+// ====================================================
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.Xml;
+
+public class HeadlineGenerator 
+{
+    protected Action<string> updatedHeadline;
```

Comment:
you may want to make `updatedHeadline `an event, so 
`public event Action<string> updatedHeadline`
that way you can remove the Register and Unregister functions on line 55